cli/cli/commands/run/run.go

```diff
@@ -270,35 +272,55 @@ func run(
 		return stacktrace.Propagate(errRunningKurtosis, "An error starting the Kurtosis code execution '%v'", starlarkScriptOrPackagePath)
 	}
 
-	if err := metricsClient.TrackKurtosisRun(starlarkScriptOrPackagePath, isRemotePackage, dryRun, isStandAloneScript); err != nil {
+	if err = metricsClient.TrackKurtosisRun(starlarkScriptOrPackagePath, isRemotePackage, dryRun, isStandAloneScript); err != nil {
 		//We don't want to interrupt users flow if something fails when tracking metrics
 		logrus.Warn("An error occurred tracking kurtosis run event")
 	}
 
 	errRunningKurtosis = readAndPrintResponseLinesUntilClosed(responseLineChan, cancelFunc, verbosity, dryRun)
+	var runStatusForMetrics bool
 	if errRunningKurtosis != nil {
-		servicesInEnclaveForMetrics, servicesInEnclaveForMetricsError := enclaveCtx.GetServices()
-		if servicesInEnclaveForMetricsError != nil {
-			logrus.Warn("Tried getting number of services in the enclave to log metrics but failed")
-		} else {
-			if err := metricsClient.TrackKurtosisRunFinishedEvent(starlarkScriptOrPackagePath, len(servicesInEnclaveForMetrics), runFailed); err != nil {
-				logrus.Warn("An error occurred tracking kurtosis run finished event")
-			}
-		}
-
-		// do not print the go trace in case of APIC failures
-		return nil
+		runStatusForMetrics = runFailed
+	} else {
+		runStatusForMetrics = runSucceeded
 	}
 
-	servicesInEnclaveForMetrics, servicesInEnclaveForMetricsError := enclaveCtx.GetServices()
+	servicesInEnclavePostRun, servicesInEnclaveForMetricsError := enclaveCtx.GetServices()
 	if servicesInEnclaveForMetricsError != nil {
-		logrus.Error("Tried getting number of services in the enclave to log metrics but failed")
+		logrus.Warn("Tried getting number of services in the enclave to log metrics but failed")
 	} else {
-		if err := metricsClient.TrackKurtosisRunFinishedEvent(starlarkScriptOrPackagePath, len(servicesInEnclaveForMetrics), runSucceeded); err != nil {
+		if err = metricsClient.TrackKurtosisRunFinishedEvent(starlarkScriptOrPackagePath, len(servicesInEnclavePostRun), runStatusForMetrics); err != nil {
 			logrus.Warn("An error occurred tracking kurtosis run finished event")
 		}
 	}
 
+	if errRunningKurtosis != nil {
+		// This error thrown by the APIC is not informative right now as it just tells the user to look at errors
+		// in the above log. For this reason we're ignoring it and returning nil. This is exceptional to not clutter
+		// the CLI output. We should still use stacktrace.Propagate for other errors.
+		return nil
+	}
+
+	if servicesInEnclaveForMetricsError != nil {
+		logrus.Warnf("Unable to retrieve the services running inside the enclave so their ports will not be" +
```

Comment:
is this the default behavior we want or do we want to put this port forwarding behind a flag `--forward-ports`; a potentially large enclave could lead to lots of port forwards; maybe this is me future proofing but if we have remote enclaves running we have

1. ports for APIC x2
2. ports for every service

We might quickly run through available ports locally; and maybe this is extra work that the person doesn't care about.